### PR TITLE
Index token v2 marketplace 

### PR DESCRIPTION
**Pull request by rtso**
## Testing
```
poetry run python -m processors.nft_marketplace_v2.processor -c config.yaml
```

config.yaml
```
# Devnet
chain_id: 65
grpc_data_stream_endpoint: "34.172.237.89:50051"
grpc_data_stream_api_key: <api_key>
db_connection_uri: "postgresql://postgres:postgres@localhost:5432/example"
starting_version_override: 543658
ending_version: 543658
```

### Listing place
Screenshot 2023-06-30 at 3 25 10 PM

### Listing cancel
Screenshot 2023-06-30 at 3 59 07 PM

### Listing fill
Screenshot 2023-06-30 at 3 59 54 PM

### Token offer placed
Screenshot 2023-07-09 at 3 10 49 PM

### Token offer canceled and filled
Screenshot 2023-07-09 at 3 13 59 PM

### Collection offer placed
Screenshot 2023-07-09 at 11 35 15 PM

### Collection offer canceled
Screenshot 2023-07-09 at 11 55 33 PM

### Collection offer filled
Screenshot 2023-07-09 at 11 57 40 PM

## Auction placed and bid placed 
Screenshot 2023-07-10 at 12 36 50 PM

### CODE REVIEW

File: python/processors/nft_marketplace_v2/constants.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from enum import Enum
+
+MARKETPLACE_V2_ADDRESS = (
+    "0xb11affd5c514bb969e988710ef57813d9556cc1e3fe6dc9aa6a82b56aee53d98"
+)
```
Comment (bowenyang007) [inline python/processors/nft_marketplace_v2/constants.py:5]: this should be in a config.

File: python/processors/nft_marketplace_v2/marketplace_v2_parser.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+            is_deleted=False,
+            marketplace=MarketplaceName.EXAMPLE_V2_MARKETPLACE.value,
+            contract_address=transaction_metadata.contract_address,
+            entry_function_id_str=transaction_metadata.entry_function_id_str_short,
+            last_transaction_version=transaction.version,
+            last_transaction_timestamp=transaction_metadata.transaction_timestamp,
+        )
+        listing = NFTMarketplaceListing(
+            transaction_version=transaction.version,
+            index=index,
```
Comment (bowenyang007) [inline python/processors/nft_marketplace_v2/marketplace_v2_parser.py:251]: you can't mix event_index with wsc_index. What if there are 2 entries and they're the same?

File: python/processors/nft_marketplace_v2/constants.py (hunk 1 above)
Comment (rtso) [reply python/processors/nft_marketplace_v2/constants.py]: What's wrong with hardcoding this? The other token v1 marketplace processor parses a ton of marketplaces and if we had those addresses in config, that'd explode the # of config vars.

File: python/processors/nft_marketplace_v2/marketplace_v2_parser.py (hunk 2 above)
Comment (rtso) [reply python/processors/nft_marketplace_v2/marketplace_v2_parser.py]: Yeah if using event_index this should be negative

File: python/processors/nft_marketplace_v2/models/nft_marketplace_listings_models.py [hunk 3]
```diff
@@ -34,16 +32,19 @@ class CurrentNFTMarketplaceListing(Base):
     inserted_at: InsertedAtType
 
 
-class NFTMarketplaceListing(Base):
-    __tablename__ = "nft_marketplace_listings"
+class NFTMarketplaceActivities(Base):
+    __tablename__ = "nft_marketplace_activities"
```
Comment (rtso) [inline python/processors/nft_marketplace_v2/models/nft_marketplace_listings_models.py:36]: Why did you change this?

**Review by bowenyang007 — APPROVED** [review]
